res/controllers/PIONEER-DDJ-ERGO-scripts.js

```diff
@@ -0,0 +1,441 @@
+                engine.setValue(effectsStrings[deck][effect], "effect_selector", -1);
+            } else {
+                engine.setValue(effectsStrings[deck][effect], "effect_selector", 1);
+            }
+            previousEffectKnobPosition = value/5;
+        }
+        //if fx knob 1 turned
+        if (control === 0x02) {
+            engine.setValue(effectsStrings[deck][effect], "meta", value/127);
+        }
+    }
+};
+
+//this plays the samples on the samplers with a click on the samplers buttons and stores its value for the sample vol knob to manipulate it
+DDJERGO.samplers = function(channel, control, value, status, group) {
+    const deck = status-0x90;
+    if (value === 0x7F) {
+        if (engine.getParameter(group, "play") === 1) {
+            engine.setParameter(group, "cue_default", 1);
+        } else {
+            engine.setParameter(group, "play", 1);
+        }
+        //store sample numbers
+        selectedSamples[0] = (control - 0x3C)/2 + (deck)*4;
+        selectedSamples[1] = (control - 0x3C)/2 + (deck)*4;
+    }
+};
+
+//this modifies the selected sample volume with sample vol knob rotations
+DDJERGO.samplerVolume = function(channel, control, value, status) {
+    const deck = status-0xB0;
+    let newValue;
+    if (value-64 > 0) { newValue = value-128; } else { newValue = value; }
+    if (newValue > 0) {
+        script.triggerControl(samplerStrings[selectedSamples[deck]], "pregain_up");
+    }
+    if (newValue < 0) {
+        script.triggerControl(samplerStrings[selectedSamples[deck]], "pregain_down");
+    }
+};
+
+//this plays the selected sample with a sample vol knob click
+DDJERGO.samplerKnobClick = function(channel, control, value, status) {
+    const deck = status-0x90;
+    const sample = selectedSamples[deck];
+    if (value === 0x7F) {
+        if (engine.getParameter(samplerStrings[sample], "play") === 1) {
+            engine.setParameter(samplerStrings[sample], "cue_default", 1);
+        } else {
+            engine.setParameter(samplerStrings[sample], "play", 1);
+        }
+    }
+};
+
+//this changes the sample stored in the knob being shift + turned
+DDJERGO.rotateSamples = function(channel, control, value, status) {
+    const deck = status-0xB0;
+    let newValue;
+    if (value-64 > 0) { newValue = value-128; } else { newValue = value; }
+    selectedSamples[deck] += newValue;
+    //makes sure samples stay within range
+    if (selectedSamples[deck] < 0) {
+        selectedSamples[deck] += 8;
+    }
+    if (selectedSamples[deck] > 7) {
+        selectedSamples[deck] -= 8;
+    }
+};
+
+//this changes both the mic and aux pregains with the aux/mic vol knob
+DDJERGO.micAuxGain = function(channel, control, value) {
+    //weird non-linear mapping for pregain
+    if (value < 64) {
+        value = value/63;
+    } else {
+        value = 3*value/64 - 125/64;
+    }
+    engine.setValue("[Microphone]", "pregain", value);
+    engine.setValue("[Auxiliary1]", "pregain", value);
```

Comment:
Users _might_ use other inputs besides the Ergo, hence this combo might not be desired.
What about
turn: adjust Aux (first label on the controller)
Shift + turn: adjust Mic

Or, detect if only one mic or one aux is configured ([`[MicrophoneN], input_configured`](https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[MicrophoneN]-input_configured)) then pick the correct one.
Else, do Shift/unshift adjust mentioned above.
Might be a bit overkill to iterate over 4 mics and 4 aux for every knob turn, you could install callback for each mic and aux pointing to one function that iterates over the devices and stores the value (1 mic | 1 aux | none/multiple) to be used here.